src/subcommands/server_test.go

```diff
@@ -0,0 +1,162 @@
+		}
+	}
+
+	return func() {
+		// Clean up test directory
+		os.RemoveAll(testDir)
+	}
+}
+
+func setupServerAndClient(t *testing.T) (pb.SemantiflyClient, func()) {
+	startTestServer()
+
+	// Set up the client
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
+
+	conn, err := grpc.NewClient("localhost:50051", opts...)
+	if err != nil {
+		t.Fatalf("Failed to create client: %v", err)
+	}
+
+	client := pb.NewSemantiflyClient(conn)
+
+	cleanup := func() {
+		conn.Close()
+	}
+
+	return client, cleanup
+}
+
+func startTestServer() {
+	go executeStartServer([]string{"--index-path", testDir})
+
+	// Allow some time for the server to start
+	time.Sleep(100 * time.Millisecond)
+}
+
+func TestServerCommands(t *testing.T) {
+	// Start server
+	cleanupTestEnv := setupTestEnvironment(t)
+	defer cleanupTestEnv()
+
+	client, cleanupClient := setupServerAndClient(t)
+	defer cleanupClient()
+
+	// Add
+	addCtx, addCancel := context.WithTimeout(context.Background(), time.Second)
+
+	testFileData1 := &pb.ContentMetadata{
+		DataType:   0,
+		SourceType: 0,
+		URI:        filepath.Join(testDir, "test_file1.txt"),
+	}
+
+	addArgs := &pb.AddRequest{
+		AddedMetadata: testFileData1,
+		MakeCopy:  true,
+	}
+
+	_, err := client.Add(addCtx, addArgs)
+	addCancel()
+	if err != nil {
+		t.Fatalf("Failed to add test file: %v", err)
+	}
+
+	// Get
+	getCtx, getCancel := context.WithTimeout(context.Background(), time.Second)
+	defer getCancel()
+
+	getReq := &pb.GetRequest{
+		Name: filepath.Join(testDir, "test_file1.txt"),
+	}
+
+	getResp, err := client.Get(getCtx, getReq)
+	if err != nil {
+		t.Fatalf("Get failed: %v", err)
+	}
+	if *getResp.Content == "" {
+		t.Errorf("Get returned empty content")
+	}
+
+	// Update
+	updCtx, updCancel := context.WithTimeout(context.Background(), time.Second)
+	defer updCancel()
+
+	testUpdateFileData := &pb.ContentMetadata{
+		DataType:   0,
+		SourceType: 0,
+		URI:        filepath.Join(testDir, "test_file2.txt"),
+	}
+
+	updReq := &pb.UpdateRequest{
+		Name:       filepath.Join(testDir, "test_file1.txt"),
+		UpdatedMetadata:   testUpdateFileData,
+		UpdateCopy: true,
+	}
+
+	_, err = client.Update(updCtx, updReq)
+	if err != nil {
+		t.Fatalf("Update failed: %v", err)
+	}
+
+	// Search
+	searchCtx, searchCancel := context.WithTimeout(context.Background(), time.Second)
+	defer searchCancel()
+
+	searchReq := &pb.LexicalSearchRequest{
+		SearchTerm: "test",
+		TopN:       5,
+	}
+
+	searchResp, err := client.LexicalSearch(searchCtx, searchReq)
+	if err != nil {
+		t.Fatalf("LexicalSearch failed: %v", err)
+	}
+	if len(searchResp.Results) == 0 {
+		t.Errorf("LexicalSearch returned no results")
+	}
+
+	// Delete
+	delCtx, delCancel := context.WithTimeout(context.Background(), time.Second)
+	defer delCancel()
+
+	delReq := &pb.DeleteRequest{
+		DeleteCopy: true,
+		Names:   []string{filepath.Join(testDir, "test_file1.txt")},
+	}
+
+	_, err = client.Delete(delCtx, delReq)
```

Comment:
for this and other subcommands with an error message response, you should check that that field is empty

(also, add a test case that verifies that the error message field is non-empty when you deliberately trigger a bug)

Reply:
done